### PR TITLE
added NOR gate method and updated nav order

### LINKED ISSUE #384: Add a NOR gate method to the sidebar?
## Is your feature request related to a problem? Please describe.
Add a NOR gate method to the sidebar? (As a complementary feature to the current NAND gate method, since they both are universal gates).

## Would you like to work on it?
Yes

### PR DESCRIPTION
**Pull request by Dishebh**
Fixes #384 

#### Changes done:
- [x] added NOR gate method and updated nav order

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add the link(s) for all the pages changed/updated from the checks tab after checks complete)
- [x] Tried Squashing the commits into one

### CODE REVIEW

File: docs/nor_gate_method.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+---
+
+## Introduction
+[NOR Gates](https://learn.circuitverse.org/docs/universal_gates.html#NOR-gate) are universal gates. By the virtue of functional completeness, NOR Gates can be used to fully represent a given boolean expression. This simplifies the expression such that only one standard gate is used throughout.
```
Comment (Shivansh2407) [inline docs/nor_gate_method.md:20]: https://learn.circuitverse.org/docs/universal_gates.html#nor-gate
Please update the link for NOR gate.

File: docs/nor_gate_method.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+---
+
+## Introduction
+[NOR Gates](https://learn.circuitverse.org/docs/universal_gates.html#NOR-gate) are universal gates. By the virtue of functional completeness, NOR Gates can be used to fully represent a given boolean expression. This simplifies the expression such that only one standard gate is used throughout.  
+
+## Functional completeness
+Functional completeness is a property pertaining to boolean logic, which states that a functionally complete boolean operator can express all possible truth tables by representing itself in a boolean expression. That is, any given boolean expression can be completely represented by using the a functionally complete boolean operator. 
+
+For example, NOR gates can be used to implement the [NOT gate](https://learn.circuitverse.org/docs/universal_gates.html#implementing-not-gate-1), the [OR gate](https://learn.circuitverse.org/docs/universal_gates.html#implementing-or-gate-1) and the [AND gate](https://learn.circuitverse.org/docs/universal_gates.html#implementing-and-gate-1).
```
Comment (Shivansh2407) [inline docs/nor_gate_method.md:25]: -1 should be removed from all the links as they are for Nand Gates. We are implementing these gates through NOT not NAND. Please update the links.

**Comment by Shivansh2407** [comment]
Also please provide the link to the CircuitVerse simulation that you have added.

File: docs/nor_gate_method.md (hunk 2 above)
Comment (Dishebh) [reply docs/nor_gate_method.md:25]: Whoops! sorry :sweat_smile: will update it.

**Comment by Dishebh** [comment]
@Shivansh2407 here's the link to circuit simulation:
https://circuitverse.org/simulator/embed/97663

**Comment by Dishebh** [comment]
Updated links and did minor changes :+1:

**Comment by schezfaz** [comment]
Hey @Dishebh, could you also ammend the [Boolean Function](https://learn.circuitverse.org/docs/bool2.html#methods-to-simplify-the-boolean-function) page and add the NOR gate method as a part of the list so as to maintain the context? thanks!

**Comment by Shivansh2407** [comment]
Hey, in the circuit B is constantval while it should be Input ?
Also you have completely used the Nand gate page language and changed some Words. Don't you think we should change the language? The heading should be same but the content should be a bit different. Please Update that accordingly.

**Comment by Dishebh** [comment]
> Hey @Dishebh, could you also ammend the [Boolean Function](https://learn.circuitverse.org/docs/bool2.html#methods-to-simplify-the-boolean-function) page and add the NOR gate method as a part of the list so as to maintain the context? thanks!

Sure, thanks for noticing!

**Comment by Dishebh** [comment]
> Small grammatical error: "any given boolean expression can be completely represented by using _**the a**_ functionally complete boolean operator. ", please edit this!

Will correct it.

**Comment by Dishebh** [comment]
> @Dishebh in the circuit, B is constantval while it should be Input ?
> Also you have completely used the Nand gate page language and changed some Words. Don't you think we should change the language? The heading should be same but the content should be a bit different. Please Update that accordingly.
> Let me know your views on this.

I was actually aiming for a different architecture since there would be some redundant text present in both the topics: `The NAND gate method` and `The NOR gate method`.

```
├── The universal gates' methods
|   ├── The NAND gate method
|   └── The NOR gate method
```

And then add the common text in both the articles under the main heading `The universal gates' methods`.

That way, it would avoid the redundancy in the docs, and make it more interactive.

**Comment by Dishebh** [comment]
But I was only hoping to do it after it got merged. What are your thoughts on it? @Shivansh2407 @schezfaz

**Comment by Dishebh** [comment]
review, please @Shivansh2407 @YashKumarVerma :)

**Comment by Dishebh** [comment]
> > @Dishebh in the circuit, B is constantval while it should be Input ?
> > Also you have completely used the Nand gate page language and changed some Words. Don't you think we should change the language? The heading should be same but the content should be a bit different. Please Update that accordingly.
> > Let me know your views on this.
> 
> I was actually aiming for a different architecture since there would be some redundant text present in both the topics: `The NAND gate method` and `The NOR gate method`.
> 
> ```
> ├── The universal gates' methods
> |   ├── The NAND gate method
> |   └── The NOR gate method
> ```
> 
> And then add the common text in both the articles under the main heading `The universal gates' methods`.
> 
> That way, it would avoid the redundancy in the docs, and make it more interactive.

The nav order hasn't been changed yet. I am planning for the above implementation when this PR gets merged, so as to avoid the possible merge conflicts thereafter. I can change the NOR gate method's content before if there's any issue of plagiarism that arises.

**Comment by YashKumarVerma** [comment]
> The nav order hasn't been changed yet. I am planning for the above implementation when this PR gets merged, so as to avoid the possible merge conflicts thereafter. I can change the NOR gate method's content before if there's any issue of plagiarism that arises.

I might be confused regarding this, but I saw a log of files being changed; their nav order. Could you help me in those a bit, please ?

**Comment by Dishebh** [comment]
@YashKumarVerma yes, I updated the nav order of all the files aligned after `the NAND gate method` so as to make `the NOR gate method` come just after the former :)

**Comment by YashKumarVerma** [comment]
> @YashKumarVerma yes, I updated the nav order of all the files aligned after `the NAND gate method` so as to make `the NOR gate method` come just after the former :)

Can i see the updated nav order ?

**Comment by Dishebh** [comment]
@YashKumarVerma here's the screenshot showing the new nav order:
Screenshot from 2020-06-12 11-50-26

You can refer the preview link in the PR description as well

**Comment by YashKumarVerma** [comment]
@Shivansh2407  what do you think about the new order ?

**Comment by Shivansh2407** [comment]
The updated nav order looks good. I would like to see both the methods under a single heading of Universal gates method after this PR is merged.

**Comment by YashKumarVerma** [comment]
Thanks a lot for your contributions.

**Pull request merged**